thirdparty/CMakeLists.txt

```diff
@@ -52,3 +52,45 @@ if(NOT thrust_POPULATED)
   # We are not using the thrust CMake targets, so no need to call `add_subdirectory()`.
 endif()
 set(THRUST_INCLUDE_DIR "${thrust_SOURCE_DIR}" PARENT_SCOPE)
+
+###################################################################################################
+# - googletest ------------------------------------------------------------------------------------
```

Comment:
googletest + googlebenchmark are indeed third party, but I wonder if they should be here, or in their respective subdirectories, since rmm does not depend on them, but vice-versa. Just wondering. I'm just learning cmake. :)

Reply:
Oh, also I was just reading about CPM recently, and wonder if that would be useful here. It would remove the need for `FetchContent_GetProperties`, change some syntax (`googletest_ADDED` vs `NOT googletest_POPULATED`), and maybe move some of the variables inline with the `CPMAddPackage` call, but I'm not sure how different it is in this situation.

Reply:
I'd say that the RMM source package does depend on gtest and gbench (if optionally) -- while the RMM library itself does not. For packages that pull in thirdparty dependencies, I'd say it's customary to do this all at once as part of the top-level cmake build, just as `find_package` usually happens there. (All the thirdparty stuff here has been consolidated into a subdirectory, but logically that's still part of what happens at the top-level, I'd argue).

It would be possible to pull in the gtest stuff only in `tests/`, and similarly for `benchmarks/`, though in practice that's more confusing, and in fact runs into trouble because the benchmarks also need gtest. The way it it is, all dependencies, ie, both dependencies that are needed for librmm (like spdlog) as well as dependencies that are only needed for build/test/etc are pulled in together, and I think that's the way it should be (and usually is).

And yes, CPM would simplify all this pulling in of thirdparty stuff, and my plan is to do that next, but in some sense it's primarily syntactic sugar for the FetchContent that's currently there (though using it enables more flexibility in the future), so I wanted to keep this separate from what's more of an actual change in this PR, ie, using gtest/gbench via `add_subdirectory()` rather than building/installing them in a separate step and then pulling in the installed version.

Reply:
Thanks for the explanation, that makes a lot of sense. 🙏